ensure_tiller_installed.go

```diff
@@ -395,6 +395,9 @@ func (c *Client) EnsureTillerInstalledWithValues(ctx context.Context, values []s
 			return microerror.Mask(err)
 		}
 	} else if !installTiller && upgradeTiller {
+		if !c.tillerUpgradeEnabled {
+			c.logger.LogCtx(ctx, "level", "debug", "message", "found an out-dated tiller but upgrades not enabled")
```

Comment:
How does this return early? Won't upgradeTiller still get called?

Reply:
🤦‍♂ Totally forgot it, Thanks to point out.